### DISCUSSION TITLE
Add `iframe` panel to integrate 3rd party apps

### DISCUSSION THREAD

**Discussion by jk-ethz** [foxglove/community#421] [Q&A]
I see Foxglove Studio becoming a powerful tool not only for visualization, but for providing a general dashboard functionality.
To that extent, it would be awesome if we could add external sites as dockable "iframe" panels. Basically a website wrapper, of which the source URL can be configured in Studio.

As an example, we are integrating [WeTTY](https://github.com/butlerx/wetty) as a web terminal solution that allows us to start and stop our nodes from the web. This would fit nicely as an embedded iframe panel/window within Foxglove Studio.

Just like that, many other applications become possible using iframes. Since we are working locally on the robot, we can fully control all required CORS headers (or even serve it from the same domain via a reverse proxy).
Is a feature like that planned?

**Comment by jtbandes** [comment]
This can be done by creating an extension: https://foxglove.dev/docs/studio/extensions/getting-started
